controllers/argocd/role.go

```diff
@@ -156,6 +181,92 @@ func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule
 	return roles, nil
 }
 
+func (r *ReconcileArgoCD) reconcileRoleForApplicationSourceNamespaces(name string, policyRules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD) ([]*v1.Role, error) {
+	var roles []*v1.Role
+
+	// create policy rules for each source namespace for ArgoCD Server
+	for _, sourceNamespace := range cr.Spec.SourceNamespaces {
+
+		namespace := &corev1.Namespace{}
+		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: sourceNamespace}, namespace); err != nil {
+			return nil, err
+		}
+
+		// do not reconcile roles for namespaces already containing managed-by label
+		// as it already contains roles with permissions to manipulate application resources
+		// reconciled during reconcilation of ManagedNamespaces
+		if value, ok := namespace.Labels[common.ArgoCDManagedByLabel]; ok {
```

Comment:
here I think we should add
```if value, ok := namespace.Labels[common.ArgoCDManagedByLabel]; ok && value != ""```
if you set managed-by = "" then operator cleans up roles effectively meaning ns is no longer managed, but still we dont create the new role if needed because the logic only checks if the label is present, not if it is actually managed